pkg/authorization/api/v1beta1/types.go

```diff
@@ -82,6 +82,40 @@ type PolicyBinding struct {
 	RoleBindings []NamedRoleBinding `json:"roleBindings"`
 }
 
+// ResourceAccessReviewSpec contains information about the action being requested.
+type ResourceAccessReviewSpec struct {
+	// Verb is one of: get, list, watch, create, update, delete
+	Verb string `json:"verb"`
+	// Resource is one of the existing resource types
+	Resource string `json:"resource"`
+	// Content is the actual content of the request for create and update
+	Content kruntime.RawExtension `json:"content,omitempty"`
+	// ResourceName is the name of the resource being requested for a "get" or deleted for a "delete"
+	ResourceName string `json:"resourceName,omitempty"`
+}
+
+// ResourceAccessReviewStatus describes who can perform the action
+type ResourceAccessReviewStatus struct {
+	// Users is the list of users who can perform the action
+	Users []string `json:"users"`
+	// Groups is the list of groups who can perform the action
+	Groups []string `json:"groups"`
+	// EvaluationError is optional.  It indicates why a ResourceAccessReview failed during evaluation
+	EvaluationError string `json:"evaluationError,omitempty"`
+}
+
+// ResourceAccessReview is a means to request a list of which users and groups are authorized to perform the
+// action specified by spec
+type ResourceAccessReview struct {
+	kapi.TypeMeta   `json:",inline"`
+	kapi.ObjectMeta `json:"metadata,omitempty" `
```

Comment:
What I'm pointing out is that ResourceAccessReview isn't an *object* in the Kubernetes API sense, so it shouldn't have ObjectMeta.  It's more like Binding, which is an operational access.  So you should nuke ObjectMeta.  Also, the "spec" / "status" distinction is mostly reserved for true "cluster entities" where you have a running resource in the cluster that has both a desired state and an actual state.  That's not the case for ResourceAccessReview.